src/docs/release/breaking-changes/actions-api-revision.md

```diff
@@ -0,0 +1,311 @@
+   to the Action).
+1) The `isEnabled` method was moved from the `Intent` class to the `Action`
+   class.
+1) The `FocusNode` argument to `Action.invoke` and `Actions.invoke` methods was removed.
+1) Invoking an action no longer creates a new instance of the `Action`.
+1) The `LocalKey` argument to the `Intent` constructor was removed.
+1) The `LocalKey` argument to `CallbackAction` was removed.
+1) The `Action` class is now a generic (`Action<T extends Intent>`) for better type
+   safety.
+1) The `OnInvokeCallback` used by `CallbackAction` no longer takes a `FocusNode`
+   argument.
+1) The `ActionDispatcher.invokeAction` signature has changed to not accept an
+   optional `FocusNode`, but instead take an optional `BuildContext`.
+1) The `LocalKey` static constants (named key by convention) in `Action` subclasses
+   have been removed.
+1) The `Action.invoke` and `ActionDispatcher.invokeAction` methods now return the
+   result of invoking the action as an `Object`.
+1) The `Action` class may now be listened to for state changes.
+1) The `ActionFactory` typedef has been removed, as it is no longer used.
+
+## Example Analyzer Failures
+
+Here are some example analyzer failures that might be encountered where an
+outdated use of the Actions API might be the cause of the problem. The specifics
+of the error might differ, and there may be other failures caused by these
+changes.
+
+```
+error: MyActionDispatcher.invokeAction' ('bool Function(Action<Intent>, Intent, {FocusNode focusNode})') isn't a valid override of 'ActionDispatcher.invokeAction' ('Object Function(Action<Intent>, Intent, [BuildContext])'). (invalid_override at [main] lib/main.dart:74)
+
+error: MyAction.invoke' ('void Function(FocusNode, Intent)') isn't a valid override of 'Action.invoke' ('Object Function(Intent)'). (invalid_override at [main] lib/main.dart:231)
+
+error: The method 'isEnabled' isn't defined for the type 'Intent'. (undefined_method at [main] lib/main.dart:97)
+
+error: The argument type 'Null Function(FocusNode, Intent)' can't be assigned to the parameter type 'Object Function(Intent)'. (argument_type_not_assignable at [main] lib/main.dart:176)
+
+error: The getter 'key' isn't defined for the type 'NextFocusAction'. (undefined_getter at [main] lib/main.dart:294)
+
+error: The argument type 'Map<LocalKey, dynamic>' can't be assigned to the parameter type 'Map<Type, Action<Intent>>'. (argument_type_not_assignable at [main] lib/main.dart:418)
+```
+
+## Migration Guide
+
+Significant changes will be required to update existing code to the new API.
+
+### Actions mapping for pre-defined actions
+
+To update the action maps in the `Actions` widget for predefined actions in
+Flutter, like `ActivateAction` and `SelectAction`, etc., you'll need to:
+
+ - Update the argument type of the actions argument, and
+ - Use an instance of a specific Intent class in the Shortcuts mapping, rather than an Intent(TheAction.key) instance.
+
+Code before migration:
+```dart
```

Comment:
Add a skip tag here because once we upgrade the stable release, the old code will fail (and break the build), yes?

Reply:
Yes. Done.